src/main/java/com/google/firebase/auth/FirebaseAuth.java

```diff
@@ -53,35 +33,19 @@
  * then use it to perform a variety of authentication-related operations, including generating
  * custom tokens for use by client-side code, verifying Firebase ID Tokens received from clients, or
  * creating new FirebaseApp instances that are scoped to a particular authentication UID.
+ * 
+ * <p>TODO(micahstairs): Add getTenantManager() method.
  */
-public class FirebaseAuth {
+public class FirebaseAuth extends AbstractFirebaseAuth {
 
   private static final String SERVICE_ID = FirebaseAuth.class.getName();
 
-  private static final String ERROR_CUSTOM_TOKEN = "ERROR_CUSTOM_TOKEN";
-
-  private final Object lock = new Object();
-  private final AtomicBoolean destroyed = new AtomicBoolean(false);
-
-  private final FirebaseApp firebaseApp;
-  private final Supplier<FirebaseTokenFactory> tokenFactory;
-  private final Supplier<? extends FirebaseTokenVerifier> idTokenVerifier;
-  private final Supplier<? extends FirebaseTokenVerifier> cookieVerifier;
-  private final Supplier<? extends FirebaseUserManager> userManager;
-  private final JsonFactory jsonFactory;
-
   private FirebaseAuth(Builder builder) {
-    this.firebaseApp = checkNotNull(builder.firebaseApp);
-    this.tokenFactory = threadSafeMemoize(builder.tokenFactory);
-    this.idTokenVerifier = threadSafeMemoize(builder.idTokenVerifier);
-    this.cookieVerifier = threadSafeMemoize(builder.cookieVerifier);
-    this.userManager = threadSafeMemoize(new Supplier<FirebaseUserManager>() {
-      @Override
-      public FirebaseUserManager get() {
-        return new FirebaseUserManager(firebaseApp);
-      }
-    });
-    this.jsonFactory = firebaseApp.getOptions().getJsonFactory();
+    super(
+        checkNotNull(builder.firebaseApp),
```

Comment:
Do `checkNotNull` in the parent constructor.

Reply:
Done.